### PR TITLE
Kl factory - deposit tests

### LINKED ISSUE #172: Port deposit integration test to shared bridge PR
### 🌟 Feature Request

The integration test we wrote for deposits should be ported to the shared bridge PR. It should pass for both the eth and custom base token setups

### PR DESCRIPTION
**Pull request by juan518munoz**
Closes #172 

## What ❔

Add native/base token `deposit` tests.

## Why ❔

Verify correct functionality. To check, `cd` into `ts-integration` directory and run the following command:
```bash
yarn test native-erc20.test.ts
```

(There are two `TODO`s that should be addressed before merging)

## Checklist

- [x] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [x] Tests for the changes have been added / updated.
- [x] Code has been formatted via `zk fmt` and `zk lint`.

### CODE REVIEW

File: core/tests/ts-integration/src/env.ts [hunk 1]
```diff
@@ -114,6 +115,13 @@ export async function loadTestEnvironment(): Promise<TestEnvironment> {
             decimals: weth.decimals,
             l1Address: weth.address,
             l2Address: l2WethAddress
+        },
+        baseToken: {
+            name: baseToken.name,
+            symbol: baseToken.symbol,
+            decimals: baseToken.decimals,
+            l1Address: baseToken.address,
+            l2Address: '0x000000000000000000000000000000000000800a' // todo: remove hardcoded value
```
Comment (fkrause98) [inline core/tests/ts-integration/src/env.ts:124]: You can use L2_ETH_TOKEN_ADDRESS here.

File: core/tests/ts-integration/src/test-master.ts [hunk 2]
```diff
@@ -64,7 +64,12 @@ export class TestMaster {
             this.l2Provider.pollingInterval = 5000;
         }
 
-        this.mainWallet = new zksync.Wallet(suiteWalletPK, this.l2Provider, this.l1Provider);
+        // TODO: suiteWalletPK should be the same as the mainWalletPK used in context-owner.ts, otherwise, account won't have funds to operate with.
```
Comment (fkrause98) [inline core/tests/ts-integration/src/test-master.ts:67]: Care to explain bit more about this comment?
Do we have an issue for this?

File: core/tests/ts-integration/src/env.ts [hunk 3]
```diff
@@ -80,6 +80,7 @@ export async function loadTestEnvironment(): Promise<TestEnvironment> {
         token = tokens[0];
     }
     const weth = tokens.find((token: { symbol: string }) => token.symbol == 'WETH')!;
+    let baseToken = tokens.find((token: { symbol: string }) => token.symbol == 'BAT')!;
```
Comment (fkrause98) [inline core/tests/ts-integration/src/env.ts:83]: Can we use const here?

Comment (juan518munoz) [reply core/tests/ts-integration/src/env.ts]: Yes, replaced it.

File: core/tests/ts-integration/src/env.ts (hunk 1 above)
Comment (juan518munoz) [reply core/tests/ts-integration/src/env.ts]: Indeed, changed it.

File: core/tests/ts-integration/src/test-master.ts (hunk 2 above)
Comment (juan518munoz) [reply core/tests/ts-integration/src/test-master.ts:67]: Upon further inspecting how the testing works, it turns out that the individual wallet used for each test wasn't getting native erc20 funds needed, this is now done inside of `distributeL1BaseToken()`.

**Pull request closed without merging**